### ISSUE TITLE
Internal error: Unexpected constructor ‘NiceFunClause’

### ISSUE THREAD

**Issue by cxandru** [msuperdock/agda-unused#1]
```
Categories/Functor/Properties.agda:127,11-28
Internal error: Unexpected constructor ‘NiceFunClause’.
```

I get this error when running `agda-unused` on https://github.com/agda/agda-categories/blob/b4da48f13545646422a0e0475c33b80e1dec1ff2/src/Categories/Functor/Duality.agda. The offending location is: https://github.com/agda/agda-categories/blob/b4da48f13545646422a0e0475c33b80e1dec1ff2/src/Categories/Functor/Properties.agda#L127

I guess what we see here is a record module instance application? Which is not supported. What I'm wondering is: Couldn't unsupported language features just return a "partial unused" result? Instead of crashing the whole process. Also, what is the reason these language features are not supported? Maybe it would be possible to contribute support for them.

**Comment by cxandru** [comment]
pinging this issue

**Comment by msuperdock** [comment]
Thanks for the issue, and sorry for the delay.

This issue is resolved by c469254, along with the two commits before it. Now `Categories.Functor.Properties` checks correctly and shows a number of unused items. (It seems there is a separate problem now when checking `Categories.Functor.Duality` related to module naming, feel free to open another issue on that.)

"Internal error" means that a value of one of Agda's internal datatypes had a form that the code assumes it can't have; in this case, I wrongly assumed that the `NiceFunClause` constructor couldn't appear within a let-clause. These internal errors should ideally never happen and are always a bug when they do, so please report if you find other ones!

On the other hand, the "unsupported" language features are features the code just doesn't yet support, mainly due to the fact that I don't use them in my own code, and that I've had some trouble figuring out how they work and how they should be checked--otherwise, I think support for these features should be easy to add. I agree that a warning would be better than a crash, but this change would take some time and my preference would be to add support for them. If you know how one of these unsupported features works and want to help me add support for it, the best way forward would be to open another issue, where we can discuss how the feature works and how it should be checked.
